app/assets/javascripts/controllers/network_router/network_router_form_controller.js

```diff
@@ -66,12 +68,25 @@ ManageIQ.angular.app.controller('networkRouterFormController', ['$http', '$scope
       .catch(miqService.handleFailure);
   };
 
+  $scope.filterCloudNetworkChanged = function(id) {
+    miqService.sparkleOn();
+    $http.get('/network_router/network_router_subnets_by_network/' + id)
```

Comment:
@gildub Thanks for creating the issue. We can work on this once this PR is merged.